common/src/main/java/org/opensearch/ml/common/connector/ConnectorAction.java

```diff
@@ -185,6 +198,103 @@ public static ConnectorAction parse(XContentParser parser) throws IOException {
+     * @return - returns the corresponding remote server name for url, if server is not in the pre-defined list,
+     * it returns null
+     */
+    public static String getRemoteServerFromURL(String url) {
+        return SUPPORTED_REMOTE_SERVERS_FOR_DEFAULT_ACTION_TYPES.stream().filter(url::contains).findFirst().orElse("");
+    }
+
+    private boolean isInBuiltProcessFunction(String processFunction) {
+        return (processFunction != null && processFunction.startsWith(INBUILT_FUNC_PREFIX));
+    }
+
+    private void validatePreProcessFunctions(String remoteServer) {
+        switch (remoteServer) {
+            case OPENAI:
+                if (!preProcessFunction.contains(OPENAI)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(OPENAI, "PreProcessFunction"));
+                }
+                break;
+            case COHERE:
+                if (!preProcessFunction.contains(COHERE)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(COHERE, "PreProcessFunction"));
+                }
+                break;
+            case BEDROCK:
+                if (!preProcessFunction.contains(BEDROCK)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(BEDROCK, "PreProcessFunction"));
+                }
+                break;
+            case SAGEMAKER:
+                if (!(TEXT_DOCS_TO_DEFAULT_EMBEDDING_INPUT.equals(preProcessFunction)
+                    || TEXT_SIMILARITY_TO_DEFAULT_INPUT.equals(preProcessFunction))) {
+                    throw new IllegalArgumentException(
+                        "LLM service is "
+                            + SAGEMAKER
+                            + ", so PreProcessFunction should be "
+                            + TEXT_DOCS_TO_DEFAULT_EMBEDDING_INPUT
+                            + " or "
+                            + TEXT_SIMILARITY_TO_DEFAULT_INPUT
+                    );
+                }
+        }
+    }
+
+    private void validatePostProcessFunctions(String remoteServer) {
+        switch (remoteServer) {
+            case OPENAI:
+                if (!postProcessFunction.contains(OPENAI)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(OPENAI, "PostProcessFunction"));
+                }
+                break;
+            case COHERE:
+                if (!postProcessFunction.contains(COHERE)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(COHERE, "PostProcessFunction"));
+                }
+                break;
+            case BEDROCK:
+                if (!postProcessFunction.contains(BEDROCK)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(BEDROCK, "PostProcessFunction"));
+                }
+                break;
+            case SAGEMAKER:
+                if (!(DEFAULT_EMBEDDING.equals(postProcessFunction) || DEFAULT_RERANK.equals(postProcessFunction))) {
```

Comment:
How can we make this more generic? Every time if we add any new type, seems like we need to update this validation?